Ensure remainder of advanced indexing backwards is contiguous.

gchanan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#36957 Ensure remainder of advanced indexing backwards is contiguous.**

This is attempt to make the current master behavior the same as on 1.4.  I don't know if this is more performant than just making the linearIndex contiguous.

Fixes: https://github.com/pytorch/pytorch/issues/36956

Differential Revision: [D21144107](https://our.internmc.facebook.com/intern/diff/D21144107)

gchanan committed:
Ensure remainder of advanced indexing backwards is contiguous.

This is attempt to make the current master behavior the same as on 1.4.  I don't know if this is more performant than just making the linearIndex contiguous.

Fixes: https://github.com/pytorch/pytorch/issues/36956

[ghstack-poisoned]

gchanan committed:
Update on "Ensure remainder of advanced indexing backwards is contiguous."


This is attempt to make the current master behavior the same as on 1.4.  I don't know if this is more performant than just making the linearIndex contiguous.

Fixes: https://github.com/pytorch/pytorch/issues/36956

Differential Revision: [D21144107](https://our.internmc.facebook.com/intern/diff/D21144107)

[ghstack-poisoned]

ngimel reviewed (approved):
This should be more performant than making linearIndex contiguous, so it's good.

albanD reviewed (approved):
Looks good !

gchanan committed:
Update on "Ensure remainder of advanced indexing backwards is contiguous."


This is attempt to make the current master behavior the same as on 1.4.  I don't know if this is more performant than just making the linearIndex contiguous.

Fixes: https://github.com/pytorch/pytorch/issues/36956

Differential Revision: [D21144107](https://our.internmc.facebook.com/intern/diff/D21144107)

[ghstack-poisoned]
